apps/gdalalg_raster_contour.cpp

```diff
@@ -0,0 +1,221 @@
+/******************************************************************************
+ *
+ * Project:  GDAL
+ * Purpose:  gdal "raster contour" subcommand
+ * Author:   Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ ******************************************************************************
+ * Copyright (c) 2025, Alessandro Pasotti <elpaso at itopen dot it>
+ *
+ * SPDX-License-Identifier: MIT
+ ****************************************************************************/
+
+#include "gdalalg_raster_contour.h"
+
+#include "cpl_conv.h"
+#include "gdal_priv.h"
+#include "gdal_utils.h"
+#include "gdal_alg.h"
+#include "gdal_utils_priv.h"
+
+//! @cond Doxygen_Suppress
+
+#ifndef _
+#define _(x) (x)
+#endif
+
+/************************************************************************/
+/*          GDALRasterContourAlgorithm::GDALRasterContourAlgorithm()    */
+/************************************************************************/
+
+GDALRasterContourAlgorithm::GDALRasterContourAlgorithm(
+    bool openForMixedRasterVector)
+    : GDALAlgorithm(NAME, DESCRIPTION, HELP_URL), m_outputLayerName("contour"),
+      m_elevAttributeName(""), m_amin(""), m_amax(""), m_levels{}
+{
+
+    AddProgressArg();
+    AddOutputFormatArg(&m_outputFormat)
+        .AddMetadataItem(GAAMDI_REQUIRED_CAPABILITIES,
+                         {GDAL_DCAP_RASTER, GDAL_DCAP_CREATECOPY});
+    AddOpenOptionsArg(&m_openOptions);
+    AddInputFormatsArg(&m_inputFormats)
+        .AddMetadataItem(GAAMDI_REQUIRED_CAPABILITIES, {GDAL_DCAP_RASTER});
+    AddInputDatasetArg(&m_inputDataset, openForMixedRasterVector
+                                            ? GDAL_OF_RASTER | GDAL_OF_VECTOR |
+                                                  GDAL_OF_MULTIDIM_RASTER
+                                            : GDAL_OF_RASTER);
+    AddOutputDatasetArg(&m_outputDataset, GDAL_OF_RASTER);
+    AddCreationOptionsArg(&m_creationOptions);
+
+    // gdal_contour specific options
+    AddArg("band", 'b', _("Specify input band number."), &m_band).SetDefault(1);
+    AddLayerNameArg(&m_outputLayerName).AddAlias("nln");
+    AddArg("aelev", 'a', _("Name of the elevation field."),
+           &m_elevAttributeName);
+    AddArg("amin", 0, _("Name of the minimum elevation field."), &m_amin);
+    AddArg("amax", 0, _("Name of the maximum elevation field."), &m_amax);
+    AddArg("3d", 0, _("Force production of 3D vectors instead of 2D."), &m_3d);
+    AddArg("inodata", 0, _("Ignore nodata values in the input raster."),
+           &m_iNodata);
+    AddArg("snodata", 0, _("Set the nodata value for the output vector."),
+           &m_sNodata);
```

Comment:
The description is incorrect. gdal_contour doc says "Input pixel value to treat as "nodata"
I guess we should call it "srcnodata" for clarity
But I'm not sure either we need it at all for now, as that could be seen as a pre-processing step to apply.

Reply:
renamed but kept